### PR TITLE
Cleanup old preview code

### PR DESCRIPTION
**Pull request by rullzer**
With https://github.com/nextcloud/gallery/pull/187 gallery also does not rely on the private class anymore.

* Old functioncall now converts to new call
* Remove \OC\Preview
* Remove \OC\Preview tests
* Remove old Preview hooks

### CODE REVIEW

**Comment by rullzer** [comment]
Also CC: @oparoz

**Review by MorrisJobke — APPROVED** [review]
Tested and works

**Review by icewind1991 — APPROVED** [review]

File: lib/public/IPreview.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ public function hasProviders();
 	 * @param boolean $scaleUp Scale smaller images up to the thumbnail size or not. Might look ugly
 	 * @return \OCP\IImage
 	 * @since 6.0.0
-	 * @deprecated 9.2.0 Use getPreview
+	 * @deprecated 11 Use getPreview
```
Comment (ChristophWurst) [inline lib/public/IPreview.php]: 12?

Comment (rullzer) [inline lib/public/IPreview.php]: Nah it was in 11. But I made that comment before we changed the internal numbering ;)

Comment (ChristophWurst) [inline lib/public/IPreview.php]: hm but that's wrong now isn't it?

Comment (MorrisJobke) [inline lib/public/IPreview.php]: Why? 9.0 -> 9, 9.1 -> 10, 9.2 -> (our) 11

Comment (MorrisJobke) [inline lib/public/IPreview.php]: See https://github.com/nextcloud/server/blob/stable11/lib/public/IPreview.php#L91 ;)

Comment (ChristophWurst) [inline lib/public/IPreview.php]: I don't get it. This is about deprecating a public API right? And master is on 12, not 11. This PR was against master. What am I missing here?

Comment (ChristophWurst) [inline lib/public/IPreview.php]: nvm, got it 🙈

**Pull request merged**